app/src/main/java/space/taran/arknavigator/mvp/presenter/adapter/tagsselector/TagsSelectorPresenter.kt

```diff
@@ -112,20 +118,57 @@ class TagsSelectorPresenter(
         }
     }
 
+    private fun kindByExt(extension: String): String? {
+        return when (extension) {
+            in ImageMetaExtra.ACCEPTED_EXTENSIONS -> "IMAGE"
+            in VideoMetaExtra.ACCEPTED_EXTENSIONS -> "VIDEO"
+            in DocumentMetaExtra.ACCEPTED_EXTENSIONS -> "DOCUMENT"
+            else -> null
+        }
+    }
+
     fun calculateTagsAndSelection() {
         Log.d(TAGS_SELECTOR, "calculating tags and selection")
         if (storage == null || index == null)
             return
 
         val resources = index!!.listIds(prefix)
+
+        var myResources = mutableListOf<ResourceMeta>()
+        resources.map {
+            myResources.add(index!!.getMeta(it))
+        }
```

Comment:
1) More descriptive name than `myResources` should be used.
2) This snippet of code seems to be possible as well:
```
val metaById = resources.map(index!!.getMeta(it))
```

Reply:
Actually, even better would be to use `index.listResources(prefix)`. It will give all metadata immediately and you always can extract `ResourceId`s from metadata.

Probably we should improve index interface later to provide kinds without necessity to read all metadata beforehand.

Reply:
ok replaced with 
`val metaById = index!!.listResources(prefix)`